### ISSUE TITLE
raise AttributeError

### ISSUE THREAD

**Issue by WLXM0922** [tjr16/bathy_nn_learning#4]
Hello, when I run the "test_place_recognition.py" program for the first time, I want to generate the "negative_pairs.pkl" file.
However, when calling the function "sample_negative_pairs" in the file "utils/dataset.py", the code is "_, _, train _ test _ pose, _ = pair _ dataset. Get _ train _ test (train _ idx, test _ idx)".
The prompt is as follows: the object "pair_dataset" has no attribute "get_train_test", which leads to "raise AttributeError" error.
But why click on "pair_dataset" will show that it has two attributes: "processed_dir" and "get_train_test".
Why is this happening? I would appreciate it if someone could answer my questions.

**Comment by lbhqw** [comment]
how do you slove it?

**Comment by WLXM0922** [comment]
The file "test_place_recognition.py" contains the code for generating negative point pairs, but it is commented out.

**Comment by lbhqw** [comment]
Thank you very much. I just found out. I tested the model and the results were not very good.

**Comment by WLXM0922** [comment]
These two test codes have many results. When you run the result graph, you need to cross it, and it will continue to produce the following results until all the point pairs are matched.

**Comment by lbhqw** [comment]
Yes, I tried it, and I found 132 sets of results, how do you judge whether it's a good match, is it the best set of results

**Comment by WLXM0922** [comment]
> Yes, I tried it, and I found 132 sets of results, how do you judge whether it's a good match, is it the best set of results

Hello, the best result of my location identification program is "mean error of 504 pairs = 22.70m", and then the best result of my feature matching program is "mean error of 20 pairs = 8.746m". What about you?
